cpp_exercise/src/lib/linuxipclib.h

```diff
@@ -0,0 +1,109 @@
+#ifndef LINUXIPCLIB_H
+#define LINUXIPCLIB_H
+
+#include <map>
+#include <string>
+#include <vector>
+#include <iostream>
+#include <map>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <cstring>
+#include <sys/types.h>
+#include <vector>
+#include <getopt.h>
+#include "ipcpipelib.h"
+
+
+namespace ipcprt {
+
+    enum Protocol {
+        invalid,
+        pipe,
+        shm,
+        msgqueue
+    };
+}
+
+ 
+
+/**
+ *      IPC Class
+ */
+
+class IPC{
+    
+    protected:
+    IPC(){};
+    virtual ~IPC(){};
+    virtual void ipcpipe(const std::string & filename)=0;
+    virtual void ipcshm()=0;
```

Comment:
Other protocols will not use filename?

Reply:
They do but becuz I working only on pipe for this time I did not change the other functions